components/handlers/module_overview.py

```diff
@@ -119,7 +120,8 @@ def GET(self):
 
         # Get module's name, description, MC and status
         module_info = model.get_module(module_code)
-
+        if not model.is_existing_module(module_code):
```

Comment:
model.validate_input(web.input(), ["code"]) is already checking whether the module exists.
This line is doing the exact same thing, so should be removed.

Also, when it's a GET method, should use web.error (return 404 page) instead of using Outcome.POST() to show error message.